ERCS/erc-7555.md

```diff
@@ -0,0 +1,211 @@
+```=
+https://<PROVIDER_URI>/sendTransaction/?
+    redirect_uri=<YOUR_REDIRECT_URI>
+    &chain_id=<CHAIN_ID>
+    &transaction=<TRANSACTION_DATA>
+```
+##### Response Syntax
+```=
+https://<YOUR_REDIRECT_URI>/sendTransaction/?
+    smart_account_address=<SMART_ACCOUNT_ADDRESS>
+    &tx_hash=<TX_HASH>
+```
+
+## Rationale
+### Redirects
+Taking inspiration from how SSO functions in the web today. We implement a similar redirect pattern, consisting of a simple request/response.
+
+#### Application
+##### Initial Request
+An application would redirect a user to a specified provider, only passing along the callback url information. This is to ensure the providers website can remain stateless, and not rely on web requests.
+##### Response from provider
+When a user is redirected to the application, it can parse the response for a signer address, and associated smart account address.
+
+#### Provider
+Upon a user navigating to the provider website, the provider would parse the redirect url and authenticate the user. The authentication method does not matter, such that it can produce a valid public address, and recover any smart accounts that may have been deployed through the provider.
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Reference Implementation
+Using `location.replace()` vs `location.href` is up to the application to decide how they wish the experience to be handled.
+
+Sample URI Request
+```=
+https://eth-sso.ethereum.org/auth?redirect_uri=http://myapp.com/eth-sso/callback/&chain_id=1
+```
+Sample Response
+```=
+http://myapp.com/callback/?smart_account_address=0xb...c
+```
+
+Application logic
+```javascript=
+// https://myapp.com
+// User triggered authentication function
+function auth() {
+    window.location.replace("https://eth-sso.ethereum.org/auth?redirect_uri=myapp.com&chain_id=1/eth-sso/callback/");
+};
+
+// App level routing logic (generic router)
+route("/eth-sso/callback/", function() {
+    let params = (new URL(document.location)).searchParams;
+    let smartAccountAddress = params.get("smart_account_address");
+});
+```
+
+Provider Logic
+```javascript=
+// eg: https://eth-sso.ethereum.org/auth
+route("/eth-sso/callback/", function("/auth") {
+    let params = (new URL(document.location)).searchParams;
+    let redirectUrl = params.get("redirect_uri");
+    // Authenticate the user (eg: with passkeys)
+    let address = "...";
+    // Get smart account if available
+    let smartAccountAddress = getSmartAccount(address);
+    window.location.replace(`http://${redirectUrl}/?smart_account_address=${smartAccountAddress}`);
+});
+```
+
+## Security Considerations
+
+<!-- Needs discussion. -->
+- Is there a concern that a user can spoof another persons address, and that could be malicious? For example, circumventing the provider, and manually calling the redirect_url with a chosen address. A way around this would be having the user actually sign a challenge message, perhaps leveraging SIWE.
```

Comment:
afaik SIWE doesn't supper eip 6492 atm which seems potentially problematic for 4337 accounts

having said that, I don't see how manually calling the `redirect_url` with a spoofed address could accomplish anything malicious. Ultimately dispatching a txn depends on the provider `/sendTransaction` endpoint which can't be spoofed unless I'm missing something

Reply:
Not sure, was set as an open question last time I had a group call about this